torch/hub.py

```diff
@@ -545,15 +549,12 @@ def _load_local(hubconf_dir, model, *args, **kwargs):
         >>> path = '/some/local/path/pytorch/vision'
         >>> model = _load_local(path, 'resnet50', pretrained=True)
     """
-    sys.path.insert(0, hubconf_dir)
-
-    hubconf_path = os.path.join(hubconf_dir, MODULE_HUBCONF)
-    hub_module = _import_module(MODULE_HUBCONF, hubconf_path)
-
-    entry = _load_entry_from_hubconf(hub_module, model)
-    model = entry(*args, **kwargs)
+    with _add_to_sys_path(hubconf_dir):
+        hubconf_path = os.path.join(hubconf_dir, MODULE_HUBCONF)
+        hub_module = _import_module(MODULE_HUBCONF, hubconf_path)
 
-    sys.path.remove(hubconf_dir)
+        entry = _load_entry_from_hubconf(hub_module, model)
+        model = entry(*args, **kwargs)
```

Comment:
We can probably move this out of the context manager, given that `torch.hub.help` also uses `_load_entry_from_hubconf` after the path is already reset (L446).